Cabal/doc/nix-local-build.markdown

```diff
@@ -0,0 +1,364 @@
+in a project; this behavior is different than that of `cabal install`,
+which applies flags to every package that would be built.  The
+motivation for this is to avoid an innocuous addition to the flags
+of a package resulting a rebuild of every package in the store
+(which might need to happen if a flag actually applied to every
+transitive dependency).  To apply options to an external package,
+use a `package` stanza in a `cabal.project` file.
+
+## cabal new-build ##
+
+`cabal new-build` takes a set of targets and builds them.  It
+automatically handles building and installing any dependencies
+of these targets.
+
+A target can take any of the following forms:
+
+* A package target: `package`, which specifies that all
+  enabled components of a package to be built.  By default,
+  test suites and benchmarks are *not* enabled, unless
+  they are explicitly requested (e.g., via `--enable-tests`.)
+
+* A component target: `[package:][ctype:]component`, which
+  specifies a specific component to be built.
+
+In component targets, `package:` and `ctype:` (valid component
+types are `lib`, `exe`, `test` and `bench`) can be used to
+disambiguate when multiple packages define the same component,
+or the same component name is used in a package (e.g., a package
+`foo` defines both an executable and library named `foo`).
+We always prefer interpreting a target as a package name rather than as
+a component name.
+
+Some example targets:
+
+~~~~
+cabal new-build lib:foo-pkg       # build the library named foo-pkg
+cabal new-build foo-pkg:foo-tests # build foo-tests in foo-pkg
+~~~~
+
+(There is also syntax for specifying module and file targets,
+but it doesn't currently do anything.)
+
+Beyond a list of targets, `cabal new-build` accepts all the
+flags that `cabal new-configure` takes; this flags are taken
+into consideration when building local packages.
+
+## cabal new-repl ##
+
+`cabal new-repl TARGET` loads all of the modules of the target
+into GHCi as interpreted bytecode.  It takes the same flags
+as ``cabal new-build``.
+
+Currently, it is not supported to pass multiple targets to ``new-repl``
+(``new-repl`` will just successively open a separate GHCi session
+for each target.)
+
+## cabal new-freeze ##
+
+`cabal new-freeze` writes out a `cabal.project.freeze` file
+which records all of the versions and flags which that are
+picked by the solver under the current index and flags.
+A `cabal.project.freeze` file has the same syntax as `cabal.project`
+and look something like this::
+
+~~~~
+constraints: HTTP ==4000.3.3,
+             HTTP +warp-tests -warn-as-error -network23 +network-uri -mtl1 -conduit10,
+             QuickCheck ==2.9.1,
+             QuickCheck +templatehaskell,
+             ...
+~~~~
+
+For end-user applications, you might consider distributing the
+`cabal.project.freeze` file in your source repository so that
+all contributors see a consistent set of dependencies.
```

Comment:
We might want to say this is highly recommended, and clarify that it's really useful for those cases where you have several people developing a single application.